### PR TITLE
Get rid of automatic competition posts.

### PR DESCRIPTION
**Pull request by AlbertoPdRF**
:warning: This PR can't be merged until all past competitions have the fields added on #4529 filled.

### CODE REVIEW

**Comment by AlbertoPdRF** [comment]
Any chance I can get some reviews so this can be pushed over the finish line? :slightly_smiling_face:

File: webroot/results/admin/scripts/check_comp_data.php [hunk 1]
```diff
@@ -150,27 +150,10 @@
...
-             Post the
-             <a data-competition-id='$compIdUrl' href='#' id='post-results-link' target='_blank' class='link-external external'>
-               results announcement
-             </a>
-             <select id='main-event-select'>
-               <option value=''>No main event</option>
-               ${eventOptionsStr}
-             </select>
-           </li>
+           <li><a href='/competitions/$compIdUrl/edit/admin' target='_blank' class='link-external external'>Post the results announcement</a></li>
```
Comment (jonatanklosko) [inline webroot/results/admin/scripts/check_comp_data.php]: Side note: while looking on this I realized that now it's less "posting results announcement" and more like "sending results announcement" / "announcing results", but I don't feel strongly about changing that ^^

File: WcaOnRails/spec/controllers/competitions_controller_spec.rb [hunk 2]
```diff
@@ -644,383 +644,61 @@
...
-          roundTypeId: "1",
-          formatId: "m",
-          value1: 4100,
-          value2: 5100,
-          value3: 6100,
-          best: 4100,
-          average: 5100,
-          regionalSingleRecord: "NAR",
-          regionalAverageRecord: "",
-        )
```
Comment (jonatanklosko) [inline WcaOnRails/spec/controllers/competitions_controller_spec.rb]: All of those removed tests could actually test the results helper that handles this logic now.

**Comment by jonatanklosko** [comment]
Superseded by #5140, thanks a lot Alberto! :shipit:

**Pull request closed without merging**